packages/core-data/src/entity-provider.js

```diff
@@ -187,16 +187,28 @@ export function useEntityBlockEditor( kind, name, { id: _id } = {} ) {
 			return EMPTY_ARRAY;
 		}
 
-		const entityRecord = getEntityRecord( kind, name, id );
-		let _blocks = parsedBlocksCache.get( entityRecord );
+		// If there's an edit, cache the parsed blocks by the edit.
+		// If not, cache by the original enity record.
+		const edits = getEntityRecordEdits( kind, name, id );
+		const isUnedited = ! edits || ! Object.keys( edits ).length;
+		const cackeKey = isUnedited ? getEntityRecord( kind, name, id ) : edits;
+		let _blocks = parsedBlocksCache.get( cackeKey );
 
 		if ( ! _blocks ) {
 			_blocks = parse( content );
-			parsedBlocksCache.set( entityRecord, _blocks );
+			parsedBlocksCache.set( cackeKey, _blocks );
```

Comment:
This might leave in memory two block lists unnecessarily. (one of entity record, one for the latest edits object). So there's a tiny memory impact, probably ok though.

Reply:
I think that's fine. Parts of this block list will be reused for edited block lists I think?